### PR TITLE
[wip] Refactored takeByTopology into smaller functions.

### PR DESCRIPTION
**Pull request by ConnorDoyle**
[wip] - new functions still need tests. old tests pass unmodified.

cc @sjenning @balajismaniam @flyingcougar

### CODE REVIEW

**Comment by sjenning** [comment]
wow! that is complicated. yet concise for what it does.  lgtm after tests :+1:

File: pkg/kubelet/cpumanager/util.go [hunk 1]
```diff
@@ -25,114 +25,147 @@ import (
...
-	}
+// Returns true if the supplied core is fully available in `topoDetails`.
+func (a *cpuAccumulator) isCoreFree(coreID int) bool {
+	return a.topoDetails.CPUsInCore(coreID).Size() == a.topo.CPUsPerCore()
+}
 
-	// 2. Acquire whole cores, if available and the container requires
-	//    at least a core's-worth of CPUs.
+// Returns socket IDs as a slice sorted by the number of whole available
+// cores, ascending
```
Comment (flyingcougar) [inline pkg/kubelet/cpumanager/util.go]: > slice sorted by the number

The only sorting here is in `AsSlice()` which sorts sockets in ascending order(by socket ID), cant see any sorting based on

> whole available cores

Missing sort from previous implementation ? 
https://github.com/ConnorDoyle/kubernetes/blob/8cf339f7842e719d0202e3bbb48c90f72ea01a25/pkg/kubelet/cpumanager/util.go#L85

Comment (ConnorDoyle) [reply pkg/kubelet/cpumanager/util.go]: Yes, unsure how I messed this up but will fix.

Comment (ConnorDoyle) [reply pkg/kubelet/cpumanager/util.go]: Actually no, I believe this is correct as written. The way the reordering looks in the diff is a little confusing. The original equivalent LOC is [here](https://github.com/ConnorDoyle/kubernetes/pull/18/files#diff-76d77ff5024ddda531805d14ea8e61b9L69). We didn't do any sorting on the list of whole sockets previously. We did sort sockets by available cores when acquiring whole cores, and that is still here (in the `freeCores` function.)

**Comment by ConnorDoyle** [comment]
Updated, added tests.

File: pkg/kubelet/cpumanager/util.go (hunk 1 above)
Comment (flyingcougar) [reply pkg/kubelet/cpumanager/util.go]: Ok, you are right, i have missed that :)

**Pull request merged**